superset-frontend/src/SqlLab/components/TableElement.tsx

```diff
@@ -17,31 +17,49 @@
  * under the License.
  */
 import React, { useState } from 'react';
-import PropTypes from 'prop-types';
 import Collapse from 'src/components/Collapse';
 import Card from 'src/components/Card';
 import ButtonGroup from 'src/components/ButtonGroup';
 import { t, styled } from '@superset-ui/core';
 import { debounce } from 'lodash';
 
 import { Tooltip } from 'src/components/Tooltip';
-import Icons from 'src/components/Icons';
 import CopyToClipboard from '../../components/CopyToClipboard';
 import { IconTooltip } from '../../components/IconTooltip';
-import ColumnElement from './ColumnElement';
+import ColumnElement, { ColumnKeyTypeType } from './ColumnElement';
 import ShowSQL from './ShowSQL';
 import ModalTrigger from '../../components/ModalTrigger';
 import Loading from '../../components/Loading';
 
-const propTypes = {
-  table: PropTypes.object,
-  actions: PropTypes.object,
-};
+interface Column {
+  name: string;
+  keys?: { type: ColumnKeyTypeType }[];
+  type: string;
+}
 
-const defaultProps = {
-  actions: {},
-  table: null,
-};
+interface Table {
+  name: string;
+  partitions?: {
+    partitionQuery: string;
+    latest: object[];
+  };
+  indexes: object[];
+  selectStar: string;
+  view: string;
+  isMetadataLoading: boolean;
+  isExtraMetadataLoading: boolean;
+  columns: Column[];
+}
+
+interface TableElementProps {
+  table: Table;
+  actions: {
+    removeDataPreview(table: Table): void;
```

Comment:
are these supposed to be functions?

Reply:
could we set these as possibly undefined as well?

Reply:
We could do that but on line 87 in the `removeTable` function it is expecting both the `removeDataPreview` and the `removeTable` actions. So I could add that they can be undefined but I would have to also add logic there to make sure they aren't undefined before invoking them. I can do that though! I just don't think it would make sense to render a table element that has a remove button that doesn't work